spyder/plugins/pylint/main_widget.py

```diff
@@ -0,0 +1,899 @@
+                    color = SUCCESS_COLOR
+                elif float(rate) > 3.:
+                    color = WARNING_COLOR
+
+                text = _("Global evaluation:")
+                text = ((text_style % (text_color, text))
+                        + (rate_style % (color, ("%s/10" % rate))))
+                if previous_rate:
+                    text_prun = _("previous run:")
+                    text_prun = " (%s %s/10)" % (text_prun, previous_rate)
+                    text += prevrate_style % (prevrate_color, text_prun)
+
+                self.treewidget.set_results(filename, results)
+                date = time.strftime("%Y-%m-%d %H:%M:%S", datetime)
+                date_text = text_style % (text_color, date)
+
+        self.ratelabel.setText(text)
+        self.datelabel.setText(date_text)
+
+    @Slot()
+    def show_log(self):
+        """
+        Show output log dialog.
+        """
+        if self.output:
+            output_dialog = TextEditor(
+                self.output,
+                title=_("Code analysis output"),
+                parent=self,
+                readonly=True
+            )
+            output_dialog.resize(700, 500)
+            output_dialog.exec_()
+
+    # --- Python Specific
+    # ------------------------------------------------------------------------
+    def get_pylintrc_path(self, filename):
+        """
+        Get the path to the most proximate pylintrc config to the file.
+        """
+        search_paths = [
+            # File"s directory
+            osp.dirname(filename),
+            # Working directory
+            getcwd_or_home(),
+            # Project directory
+            self.get_option("project_dir"),
+            # Home directory
+            osp.expanduser("~"),
+        ]
+
+        return get_pylintrc_path(search_paths=search_paths)
+
+    @Slot()
+    def select_file(self, filename=None):
+        """
+        Select filename using a open file dialog and set as current filename.
+
+        If `filename` is provided, the dialog is not used.
+        """
+        if filename is None:
+            self.sig_redirect_stdio_requested.emit(False)
+            filename, _selfilter = getopenfilename(
+                self,
+                _("Select Python file"),
+                getcwd_or_home(),
+                _("Python files") + " (*.py ; *.pyw)",
+            )
+            self.sig_redirect_stdio_requested.emit(True)
+
+        if filename:
+            self.set_filename(filename)
+            self.start_code_analysis()
+
+    def get_command(self, filename):
+        """
+        Return command to use to run code analysis on given filename
+        """
+        command_args = []
+        if PYLINT_VER is not None:
+            command_args = [
+                "-m",
+                "pylint",
+                "--output-format=text",
+                "--msg-template="
+                ""{msg_id}:{symbol}:{line:3d},{column}: {msg}"",

```

Comment:
Seems like the replace needs some changes here (this is causing the error in the CIs)

Reply:
Also, some PEP8 errors raised. I know these kind of things are kind of a pain to change so thanks for the effort :+1:

Reply:
No worries @dalthviz, it is how it goes.

Thanks for all the reviewing. You are all doing a great job, and practice makes perfect :-) !